domain/access/service/user.go

```diff
@@ -174,7 +174,7 @@ func (s *UserService) addUserWithPassword(ctx context.Context, arg AddUserArg) e
 		return errors.Trace(err)
 	}
 
-	err = s.st.AddUserWithPasswordHash(ctx, arg.UUID, arg.Name, arg.DisplayName, arg.CreatorUUID, arg.Permission, hash, salt)
+	err = s.st.AddUserWithPasswordHash(ctx, arg.UUID, arg.Name, arg.DisplayName, arg.External, arg.CreatorUUID, arg.Permission, hash, salt)
```

Comment:
Both of these should now include validation so that we can _not_ use an external user.

Reply:
I've added validation to `AddUser` in the service layer that calls through to either of the methods in state.

Reply:
Have you got something you need to push? I can't see this.

Reply:
Ah whoops, pushed now.